jobs/sweep_schedulers/optuna_scheduler/optuna_scheduler.py

```diff
@@ -484,14 +521,30 @@ def _poll_run(self, orun: OptunaRun) -> bool:
 
         orun.num_metrics = len(metrics)
 
-        # run hasn't started
-        if self._runs[orun.sweep_run.id].state.is_alive or len(metrics) == 0:
-            logger.debug(f"Run ({orun.sweep_run.id}) has no metrics")
+        # run still running
+        if self._runs[orun.sweep_run.id].state.is_alive:
             return False
 
         # run is complete
         prev_metrics = orun.trial._cached_frozen_trial.intermediate_values
-        last_value = prev_metrics[orun.num_metrics - 1]
+        last_value = (
+            -1
+            if self.study.direction == optuna.study.StudyDirection.MAXIMIZE
+            else float("inf")
+        )
+        if (
+            self._runs[orun.sweep_run.id].state == RunState.FINISHED
+            and len(prev_metrics) == 0
+        ):
+            # run finished correctly, but never logged a metric
+            wandb.termwarn(
```

Comment:
If this happens 3 times in a row just throw an error, and ignore the first 2 runs

Reply:
<img width="736" alt="Screen Shot 2023-06-16 at 2 21 26 PM" src="https://github.com/wandb/launch-jobs/assets/19414170/5b3c11b2-dcfb-400d-9de5-b0307a6eaf7e">